src/Deserializers/StatementDeserializer.php

```diff
@@ -61,15 +61,9 @@ public function __construct(
 	 * @return bool
 	 */
 	public function isDeserializerFor( $serialization ) {
-		return $this->hasType( $serialization ) && $this->hasCorrectType( $serialization );
-	}
-
-	private function hasType( $serialization ) {
-		return is_array( $serialization ) && array_key_exists( 'type', $serialization );
-	}
-
-	private function hasCorrectType( $serialization ) {
-		return in_array( $serialization['type'], array( 'claim', 'statement' ) );
+		return is_array( $serialization )
+			&& array_key_exists( 'type', $serialization )
+			&& ( $serialization['type'] === 'claim' || $serialization['type'] === 'statement' );
```

Comment:
I disagree. What puzzled me the most was the code that checked if a variable "has type" and "has correct type". What "type"? Does this check if the variable is an array? How can such a type check be "correct"?

If you find `in_array( $serialization['type'], [ 'claim', 'statement' ] )` more readable we can change the last line.